SecondClasses/rev_polish_notation/src/rev_polish_notation.erl

```diff
@@ -2,6 +2,28 @@
 
 -export([evaluate/1]).
 
+evaluate(List)->
+    R = lists:foldl(fun next/2, [], List),
+    lists:last(R).
 
-evaluate(_) ->
-    1.
\ No newline at end of file
+next(Elem, Stack) when is_integer(Elem) ->
+    [Elem | Stack];
+next(Elem, Stack) when is_function(Elem)->
+    case get_fun_args_number(Elem) of
+        Arity when Arity == 2 andalso length(Stack) >= 2 ->
+            {Last, Remaining} = get_stack_head_and_ramaining(Stack),
```

Comment:
you can just match it directly without wrapper function: 
```erlang
[Last, Remaining] = Stack,
```

Reply:
do you mean, that match would be like:
`{Last, Remaining} = [Last, Remaining] = Stack`
will it work?

Reply:
hm... it wont work of cource. Can you explain this?